slm_porosity/heat_transfer.f90

```diff
@@ -1027,6 +1052,39 @@ FUNCTION laser_heat_flux (x, nlocal, factor)
     END IF
   END FUNCTION laser_heat_flux
 
+
+  FUNCTION laser_volumetric_heat_flux (x, nlocal)
+    IMPLICIT NONE
+    REAL(pr), INTENT(IN) :: x(nlocal,dim)
+    INTEGER,  INTENT(IN) :: nlocal
+    REAL(pr) :: normalizing_factor
+    REAL(pr) :: laser_volumetric_heat_flux(nlocal)
+
+    normalizing_factor = 6*SQRT(3.0_pr)/(((pi)**1.5_pr)*laser_beam_length*laser_beam_width*laser_beam_length)
+    IF (dim.EQ.3) THEN
+      laser_volumetric_heat_flux(:) = normalizing_factor*absorptivity*laser_power* &
+        laser_gauss(x(:,1), nlocal, initial_laser_position(1) - scanning_speed*t, laser_beam_length)* &
+        laser_gauss(x(:,2), nlocal, initial_laser_position(2), laser_beam_width)* &
+        laser_gauss(x(:,3), nlocal, initial_laser_position(3), laser_beam_depth)
+    ELSE
+      laser_volumetric_heat_flux(:) = power_factor_2d*normalizing_factor*absorptivity*laser_power/&
+        (laser_beam_length*laser_beam_depth*pi)* &
+          laser_gauss(x(:,1), nlocal, initial_laser_position(1) - scanning_speed*t, laser_beam_length)* &
+          laser_gauss(x(:,2), nlocal, initial_laser_position(2), laser_beam_depth)
+    END IF
+  END FUNCTION laser_volumetric_heat_flux
+
+    ! gaussian decreasing at upper part could vary.
```

Comment:
1. Wrong indentation. 2. Comment is a) confusing, b) full of slang, c) longer than the code.